cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionCheckCommand.java

```diff
@@ -0,0 +1,253 @@
+        return cliCheckerExecutor.executeChecks().stream()
+                .filter(result -> !result.isSuccessful())
+                .collect(Collectors.toList());
+    }
+
+    String getPhabricatorMessage(String notificationText) {
+
+        ImmutableMap<String, Object> messageParamMap = ImmutableMap.<String, Object>builder()
+                .put("icon", PhabricatorIcon.WARNING.toString())
+                .put("header", "**i18n source string checks failed**")
+                .put("notificationText", notificationText)
+                .build();
+        String message = "{icon} {header}" + System.lineSeparator() + System.lineSeparator() + "{notificationText}";
+        return getFormattedPhabricatorMessage(messageTemplate, "baseMessage", getBaseMessage(messageParamMap, message));
+    }
+
+    private String buildPhabricatorComment(List<CliCheckResult> failedCheck) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("**Failed checks:**");
+        addDoubleNewLine(sb);
+        failedCheck.stream().forEach(check -> {
+            sb.append("//**" + check.getCheckName() + "**//");
+            addDoubleNewLine(sb);
+            sb.append(check.getNotificationText());
+        });
+        addDoubleNewLine(sb);
+        sb.append("**Please correct the above issues in a new commit.**");
+        return sb.toString();
+    }
+
+    private void addDoubleNewLine(StringBuilder sb) {
+        sb.append(System.lineSeparator() + System.lineSeparator());
+    }
+
+    private void checkPhabricatorPreconditions() {
+        if (diffId != null) {
+            PhabricatorPreconditions.checkNotNull(differentialDiff);
+            PhabricatorPreconditions.checkNotNull(differentialRevision);
+        }
+    }
+
+    private CliCheckerExecutor getCliCheckerExecutor(List<AssetExtractionDiff> assetExtractionDiffs) {
+        CliCheckerExecutor cliCheckerExecutor = new CliCheckerExecutor(generateChecks(assetExtractionDiffs));
+        consoleWriter.newLine().a("Running checks against new strings").println();
+        return cliCheckerExecutor;
+    }
+
+    private void outputFailuresToCommandLine(List<CliCheckResult> failedCheckNames) {
+        consoleWriter.fg(Ansi.Color.YELLOW).newLine().a("Failed checks: ").println();
+        failedCheckNames.stream().forEach(check -> {
+            consoleWriter.fg(Ansi.Color.YELLOW).newLine().a(check.getCheckName()).println();
+            consoleWriter.fg(Ansi.Color.YELLOW).newLine().a(check.getNotificationText().replaceAll("\\*", "\t*")).println();
+        });
+    }
+
+    private CliCheckerOptions generateCheckerOptions() {
+        Set<PlaceholderRegularExpressions> regexSet = generateParameterRegexSet();
+        Set<String> hardFailureSet = generateHardFailureSet();
+        return new CliCheckerOptions(regexSet, hardFailureSet, dictionaryAdditionsFilePath, glossaryFilePath);
+    }
+
+    private Set<String> generateHardFailureSet() {
```

Comment:
Knit: renaming this might be a little more clear, `getClassNamesOfHardFailures` or something along those lines